src/audio/pipeline/pipeline-params.c

```diff
@@ -162,8 +165,18 @@ static int pipeline_comp_hw_params(struct comp_dev *current,
 
 	/* Fetch hardware stream parameters from DAI component */
 	if (dev_comp_type(current) == SOF_COMP_DAI) {
-		ret = comp_dai_get_hw_params(current,
-					     &ppl_data->params->params, dir);
+		struct dai_data *dd;
+#if CONFIG_IPC_MAJOR_3
+		dd = comp_get_drvdata(current);
+#elif CONFIG_IPC_MAJOR_4
+		struct copier_data *cd = comp_get_drvdata(current);
+
+		dd = cd->dd[0];
+#else
+#error Unknown IPC major version
+#endif
+		/* dai device already removed, currently it is copier */
+		ret = dai_zephyr_get_hw_params(dd, current, &ppl_data->params->params, dir);
```

Comment:
not true for multi endpoint DAI yet

Reply:
ok, seems I put too much in one patch again, let me split and upload with new patches.

Reply:
since multi-endpoint PR merged, seems this part of change need involve again, let me try to submit a new patch to cover this.